Service/Request/Visitor/RemoveNullFieldsRequestVisitor.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace Auto1\ServiceAPIClientBundle\Service\Request\Visitor;
+
+use Http\Message\StreamFactory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Class RemoveNullFieldsRequestVisitor.
+ */
+class RemoveNullFieldsRequestVisitor implements RequestVisitorInterface
+{
+    /**
+     * @var EndpointConfiguration[]
+     */
+    private $endpoints;
+
+    /**
+     * @var StreamFactory
+     */
+    private $factory;
+
+    /**
+     * RemoveNullFieldsRequestVisitor constructor.
+     *
+     * @param StreamFactory $factory
+     * @param EndpointConfiguration[] $endpoints
+     */
+    public function __construct(
+        StreamFactory $factory,
+        array $endpoints
+    ) {
+        $this->factory = $factory;
+        $this->endpoints = $endpoints;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visit(RequestInterface $request): RequestInterface
+    {
+        if ($this->isCompatible($request)) {
+            $request = $request->withBody($this->cleanBody($request->getBody()));
+        }
+
+        return $request;
+    }
+
+    /**
+     * @return bool
+     */
+    private function isCompatible(RequestInterface $request): bool
+    {
+        foreach ($this->endpoints as $endpoint) {
+            if (
+                $request->getMethod() === $endpoint->getMethod()
+                && preg_match($endpoint->getPathRegexp(), $request->getUri()->getPath())
+            ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return StreamInterface
+     */
+    private function cleanBody(StreamInterface $requestBody): StreamInterface
+    {
+        $data = json_decode($requestBody->getContents(), true);
+        $cleanData = $this->clean($data);
+
+        return $this->factory->createStream(json_encode($cleanData));
+    }
+
+    /**
+     * @param array $input
+     *
+     * @return array
+     */
+    private function clean($input): array
+    {
+        foreach ($input as &$value) {
+            if (is_array($value)) {
+                $value = $this->clean($value);
+            }
+        }
+
+        return array_filter($input, static function ($value) {
```

Comment:
`array_filter` preserves array keys — so it means you will have `{"0" => "value", "2" => "value"}` after serialization

Reply:
I see the point, from the original ticket there is no a clear answer what to do in this case. I don't know if we are counting for cases of just array of elements without key to be null. Generally arrays of elements are object not just values.

Reply:
Since it is `auto1-oss`, we talk not only about internal usage. So, for sure, it could be a simple array of values. 
After you will remove null-elements, it still should be an array (I mean json-representation, not PHP object).